presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java

```diff
@@ -82,6 +82,9 @@ public Pattern getPattern()
             return Optional.empty();
         }
 
+        if (!child.getOrderBySymbols().isEmpty()) {
```

Comment:
I put order by symbols in MarkDistinctNode because otherwise the order by fields can be pruned by PruneUnreferencedOutputs. That's the only real use case for this.